src/System.Net.Requests/tests/RequestStreamTest.cs

```diff
@@ -107,6 +133,77 @@ public void WriteAsync_TokenIsCanceled_TaskIsCanceled()
             }
         }
 
+        #endregion
+
+        #region BeginWrite
+
+        private IAsyncResult BeginWriteWrapper(Stream stream, byte[] buffer, int offset, int count)
+        {
+            AsyncCallback callback = null;
+            object state = new object();
+            IAsyncResult result = stream.BeginWrite(buffer, offset, count, callback, state);
+            stream.EndWrite(result);
+
+            return result;
+        }
+
+        [Fact]
+        public void BeginWriteAsync_BufferIsNull_ThrowsArgumentNullException()
+        {
+            using (Stream stream = GetRequestStream())
+            {
+                Assert.Throws<ArgumentNullException>(() => BeginWriteWrapper(stream, null, 0, 1));
```

Comment:
As noted above, I don't see a need for this BeginWriteWrapper. This should just be:
```C#
Assert.Throws<ArgumentNullException>("buffer", () => stream.BeginWrite(null, 0, 1, null, null));
```
Note, too, verifying the argument name.